pkg/apiserver/logsearch/pack.go

```diff
@@ -76,24 +78,28 @@ func serveTaskForDownload(task *TaskModel, c *gin.Context) {
 		_ = c.Error(utils.ErrInvalidRequest.New("Log is not available for this task"))
 		return
 	}
-
-	f, err := os.Open(*task.LogStorePath)
-	if err != nil {
-		_ = c.Error(err)
-		return
-	}
-	stat, err := f.Stat()
-	if err != nil {
-		_ = c.Error(err)
-		return
-	}
-
-	contentType := "application/zip"
+	reader, writer := io.Pipe()
+	go func() {
+		tw := tar.NewWriter(writer)
+		defer writer.Close()
+		defer tw.Close()
+		err := dumpLog(*task.LogStorePath, tw)
+		if err != nil {
+			_ = c.Error(err)
+			return
+		}
+		if task.SlowLogStorePath != nil {
+			if err = dumpLog(*task.SlowLogStorePath, tw); err != nil {
+				_ = c.Error(err)
+				return
+			}
+		}
+	}()
+	contentType := "application/tar"
 	extraHeaders := map[string]string{
-		"Content-Disposition": fmt.Sprintf(`attachment; filename="%s"`, stat.Name()),
+		"Content-Disposition": `attachment; filename="logs.tar"`,
```

Comment:
I think the previous implementation is better, which always generates a unique name each download